<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QMainWindow" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1192</width>
    <height>814</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout_3">
    <item>
     <widget class="QTabWidget" name="tabs">
      <property name="focusPolicy">
       <enum>Qt::TabFocus</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="usesScrollButtons">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tabDraw">
       <attribute name="title">
        <string>Рисование эллипса</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <layout class="QVBoxLayout" name="drawLeftPanel">
          <item>
           <widget class="QLabel" name="drawLabelLineStart">
            <property name="text">
             <string>Координаты центра эллипса:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="drawLayoutCenter">
            <item>
             <widget class="QLineEdit" name="drawLineCenterX">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>X</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="drawLineCenterY">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Y</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="drawSpacer1">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Minimum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="drawLabelLineEnd">
            <property name="text">
             <string>Размеры малой и большой полуосей:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="drawLayoutSize">
            <item>
             <widget class="QLineEdit" name="drawLineSizeX">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>X</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="drawLineSizeY">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Y</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="drawSpacer2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="drawLabelAlgorithm">
            <property name="text">
             <string>Используемый алгоритм отрисовки:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="drawComboAlgorithm">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Библиотечный алгоритм</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Каноническое уравнение</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Параметрическое уравнение</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Алгоритм Брезенхема</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Алгоритм средней точки</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="drawSpacer3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Minimum</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="drawLabelColor">
            <property name="text">
             <string>Цвет отрезка:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="drawComboColor">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <item>
             <property name="text">
              <string>Чёрный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Белый</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Красный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Зелёный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Синий</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="drawSpacer4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>16</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="drawButtonPaint">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Отрисовать эллипс</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="drawSpacer5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="drawButtonClear">
            <property name="text">
             <string>Очистить поле</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="Canvas" name="drawCanvas">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>700</width>
              <height>700</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer_3">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabSpectrum">
       <attribute name="title">
        <string>Рисование спектра</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <layout class="QVBoxLayout" name="spectrumLeftPanel">
          <item>
           <widget class="QLabel" name="spectrumLabelCenter">
            <property name="text">
             <string>Малая и большая полуоси начальн:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="spectrumLayoutBegin">
            <item>
             <widget class="QLineEdit" name="spectrumLineBeginX">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>X</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="spectrumLineBeginY">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Y</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QLabel" name="spectrumLabelEnd">
            <property name="text">
             <string>Малая и большая полуоси конечн:</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="spectrumLayoutEnd">
            <item>
             <widget class="QLineEdit" name="spectrumLineEndX">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>X</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="spectrumLineEndY">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="placeholderText">
               <string>Y</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="spectrumSpacer1">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="spectrumLabelAmount">
            <property name="text">
             <string>Количество эллипсов</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="spectrumLineAmount">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="spectrumSpacer2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="spectrumLabelAlgorithm">
            <property name="text">
             <string>Используемый алгоритм отрисовки:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="spectrumComboAlgorithm">
            <item>
             <property name="text">
              <string>Библиотечный алгоритм</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Каноническое уравнение</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Параметрическое уравнение</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Алгоритм Брезенхема</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Алгоритм средней точки</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="spectrumSpacer3">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>8</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QLabel" name="spectrumLabelColor">
            <property name="text">
             <string>Цвет эллипса:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="spectrumComboColor">
            <item>
             <property name="text">
              <string>Чёрный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Белый</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Красный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Зелёный</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Синий</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="spectrumSpacer4">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>5</width>
              <height>16</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="spectrumButtonPaint">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Отрисовать спектр эллипсов</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="spectrumSpacer5">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="spectrumButtonClear">
            <property name="text">
             <string>Очистить поле</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="Canvas" name="spectrumCanvas">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>700</width>
              <height>700</height>
             </size>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="horizontalSpacer_4">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tabLadder">
       <attribute name="title">
        <string>График ступенчатости</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tabChart">
       <attribute name="title">
        <string>Гистограмма производительности</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Canvas</class>
   <extends>QOpenGLWidget</extends>
   <header>canvas.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>tabs</tabstop>
  <tabstop>drawLineCenterX</tabstop>
  <tabstop>drawLineCenterY</tabstop>
  <tabstop>drawLineSizeX</tabstop>
  <tabstop>drawLineSizeY</tabstop>
  <tabstop>drawComboAlgorithm</tabstop>
  <tabstop>drawComboColor</tabstop>
  <tabstop>drawButtonPaint</tabstop>
  <tabstop>spectrumLineBeginX</tabstop>
  <tabstop>spectrumLineBeginY</tabstop>
  <tabstop>spectrumComboAlgorithm</tabstop>
  <tabstop>spectrumComboColor</tabstop>
  <tabstop>spectrumButtonPaint</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
